File: computer-club/backend/src/shared/errors/unauthorized.error.ts
import { BaseError } from './base.error';
import { ErrorMessage } from '../../constants/enums';
import { HttpStatus } from '@nestjs/common';

export class UnauthorizedError extends BaseError {
  constructor(message: string = ErrorMessage.UNAUTHORIZED) {
    super(HttpStatus.UNAUTHORIZED, message);
  }
}

File: bookstore/spring-backend/src/main/kotlin/masha/bookstore/errors/UnauthorizedError.kt
package masha.bookstore.errors

import masha.bookstore.core.BaseError
import org.springframework.http.HttpStatus

class UnauthorizedError(
    override val message: String = ErrorMessage.UNAUTHORIZED
) : BaseError(
    HttpStatus.UNAUTHORIZED,
    ApiError(
        code = HttpStatus.UNAUTHORIZED,
        message = message
    )
)